server/base/src/main/java/org/apache/accumulo/server/util/time/SimpleTimer.java

```diff
@@ -100,10 +93,19 @@ static int getInstanceThreadPoolSize() {
     return instanceThreadPoolSize;
   }
 
-  private SimpleTimer(int threadPoolSize) {
+  protected SimpleTimer(int threadPoolSize) {
     executor = Executors.newScheduledThreadPool(threadPoolSize,
         new ThreadFactoryBuilder().setNameFormat("SimpleTimer-%d").setDaemon(true)
-            .setUncaughtExceptionHandler(new ExceptionHandler()).build());
+            .setUncaughtExceptionHandler(getUncaughtExceptionHandler()).build());
+  }
+
+  protected Thread.UncaughtExceptionHandler getUncaughtExceptionHandler() {
+    return new Thread.UncaughtExceptionHandler() {
+      @Override
+      public void uncaughtException(Thread t, Throwable e) {
+        log.warn("SimpleTimer task failed", e);
+      }
+    };
```

Comment:
I'm curious what non-critical use cases we still have for this, that would warrant continuing to swallow OOM errors and such.

Reply:
I agree, I'm not sure I have the ability to decide what is critical vs what is not. I came across a reference to new JVM parameters that may take OOME handling out of our hands: https://bugs.java.com/bugdatabase/view_bug.do?bug_id=8152669

Reply:
It looks like this new parameter does not work in all cases. https://bugs.openjdk.java.net/browse/JDK-8155004.